galahad/client/formats.py

```diff
@@ -73,3 +74,40 @@ def build_span_classification_document(
 
     document = Document(text=text, annotations=annotations.to_dict(), version=version)
     return document
+
+#original_doc: Document
+def build_span_classification_response(text: str, sentences: List[List[str]], spans: List[List[Span]] = None, version: int = 0) -> Document:
```

Comment:
I would pass the original doc here to which we respond, that way you do not need to search tokens (which is also slow, find is O(n).